100-level-AsyncAPI-Intro/step2.md

```diff
@@ -0,0 +1,329 @@
+  - name: chat
+    description: Operations related to chat
+```
+
+- **Components**
+
+`Component` is like the toolbox in Chan's spaceship. It contains reusable parts like message definitions and schemas that can be used across the API. Each component is linked to a particular schema, which describes the structure of the message content.
+
+```yml
+components:
+  messages:
+    ChatMessage:
+      name: ChatMessage
+      title: Chat Message
+      contentType: application/json
+      payload:
+        $ref: '#/components/schemas/ChatMessagePayload'
+  schemas:
+    ChatMessagePayload:
+      type: object
+      properties:
+        userId:
+          type: string
+          description: The ID of the user who sent the message
+        message:
+          type: string
+          description: The chat message
+```
+
+## Operations, messages, and schemas
+
+- **Operations**
+
+`Operations` in AsyncAPI are like the instructions for the spaceship. They define what actions can be performed within each channel, such as sending or receiving messages. Each operation is associated with a message and a schema.
+
+```yml
+operations:
+  sendChatMessage:
+    action: 'send'
+    summary: Send a chat message to a specific room
+    channel:
+      $ref: '#/channels/roomidchat'
+```
+
+- **Messages**
+
+`Messages` in AsyncAPI are the content exchanged during operations, just like the words Eve from Capuccinova and Chan from Brownieterra use to communicate. They describe the structure and content of the messages sent or received.
+
+
+```yml
+components:
+  messages:
+    ChatMessage:
+      name: ChatMessage
+      title: Chat Message
+      contentType: application/json
+      payload:
+        $ref: '#/components/schemas/ChatMessagePayload'
+```
+
+- **Schemas**
+
+`Schemas` in AsyncAPI are like the blueprints for the spaceship. They define the structure of the message content, ensuring that everything fits together correctly. Each schema is associated with a component and describes the structure and content of the messages.
+
+```yml
+components:
+  schemas:
+    ChatMessagePayload:
+      type: object
+      properties:
+        userId:
+          type: string
+          description: The ID of the user who sent the message
+        message:
+          type: string
+          description: The chat message
+```
+
+Now it's time to help Chan establish a sophisticated communication system with Eve on Capuccinova. With your new expertise in AsyncAPI, you'll assist him in creating an AsyncAPI document from scratch.
+
+
+#### Exercise 1: Create an AsyncAPI document
+
+
+1. In the KillerCoda terminal, create a new file by typing `nano chan-to-capuccinova.yaml`.
+
+2. Enter the following AsyncAPI document structure:
+   ```yaml
+   asyncapi: 3.0.0
+   info:
+     title: Space Communication API
+     version: '1.0.0'
+     description: API for communication between Chan's spaceship and Capuccinova
+
+   servers:
+     production:
+       host: mqtt://localhost
```

Comment:
Default MQTT (unencrypted) port is 1883.
Additionally, you don't need to specify the protocol in the host since it is already present in the `protocol` field
```suggestion
       host: localhost:1883
```